Gemfile

```diff
@@ -18,6 +18,7 @@ gem "i18n"
 gem "i18n-js"
 gem "config"
 gem "bootstrap-sass", "3.3.7"
+gem "bcrypt", "3.1.12"
```

Comment:
- Gem này dùng làm j em?
- Kể tên một số phương thức nổi bật của gem?

Reply:
- "bcrypt" là một dạng hàm hoặc thuật toán dùng để mã hóa mật khẩu, là một hàm băm 1 chiều (không thể khôi phục được chuỗi ban đầu từ chuỗi đã băm).
- Một số phương thức nổi bật của gem bcrypt:
 + Passwword.create: tạo password mới với đầu vào là chuỗi kí tự thường, kết quả là 1 chuỗi băm 
 + Password.new: tạo password với đầu vào là một chuỗi đã băm.
 + is_password? : kiểm tra password có khớp hay không, kết quả trả về true hoặc false
 + has_secure_password: để sử dụng hàm này trong db cần có trường password_digest, hàm này sẽ tạo pasword đã được băm vào trường password_digest. Hàm tạo ra hai attr ảo là password và password_confirmation để xác nhận sự hiện diện và phải khớp password.
 + authenticate: trả về user nếu nhập đúng pasword, trả về false nếu không khớp.